database_persistence.rb

```diff
@@ -17,6 +19,28 @@ def query(statement, *params)
     @db.exec_params(statement, params)
   end
 
+  def user_by_id(id)
+    sql = <<~SQL
+      SELECT
+        u.slackname,
+        u.firstname,
+        u.lastname,
+        u.about_me,
+        t.name AS track,
+        c.code AS course,
+        ti.name AS timezone
+      FROM users u
+        JOIN tracks t ON u.track_id = t.id
+        JOIN courses c ON u.course_id = c.id
+        JOIN timezones ti ON u.timezone_id = ti.id
+      WHERE u.id = $1
+    SQL
+
+    result = query(sql, id)
+    tuple = result.first
```

Comment:
I don't understand why we need to call `first` here, isn't the `query` already returning just one element (the row in db with `id` is passed to `query`, right?

Reply:
Yep, I'll delete first, but I'll have to test that the returned data type is still a tuple and an array of tuples of length one

Reply:
Just kidding! exec_params always returns a collection and therefore requires selecting an index, even if the collection has a length of one